azurerm/internal/services/containers/container_registry_token_resource_test.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type ContainerRegistryTokenResource struct{}
+
+func TestAccContainerRegistryToken_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_token", "test")
+	r := ContainerRegistryTokenResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("status").HasValue("enabled"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccContainerRegistryToken_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_token", "test")
+	r := ContainerRegistryTokenResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_container_registry_token"),
+		},
+	})
+}
+
+func TestAccContainerRegistryToken_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_token", "test")
+	r := ContainerRegistryTokenResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data, "enabled"),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("status").HasValue("enabled"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccContainerRegistryToken_completeUpdated(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_token", "test")
+	r := ContainerRegistryTokenResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.complete(data, "enabled"),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("status").HasValue("enabled"),
+			),
+		},
+		{
+			Config: r.complete(data, "disabled"),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("status").HasValue("disabled"),
+			),
+		},
+	})
+}
+
+func (t ContainerRegistryTokenResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := azure.ParseAzureResourceID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroup := id.ResourceGroup
+	registryName := id.Path["registries"]
+	name := id.Path["tokens"]
+
+	resp, err := clients.Containers.TokensClient.Get(ctx, resourceGroup, registryName, name)
```

Comment:
could we use the generated parse function here to?
```suggestion
	id, err := azure.ParseACR(state.ID)
	if err != nil {
		return nil, err
	}
	
	resp, err := clients.Containers.TokensClient.Get(ctx, id.ResourceGroup, id.RegistryName, id.Name)
```